### PR TITLE
Add Trino 400 release notes

### LINKED ISSUE #14360: Add Trino 399 release notes

### LINKED ISSUE #14509: Release notes for 400

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 400 release. Fixes #14509 

## Non-technical explanation

N/A

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

## Verification for each commit

Format: commit, committer/maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 6 Oct 2022 ✅ 

9e6208619d7a4c1f0fc3f736c5bc70c1632756a1 @arhimondr #14448 ✅ rn ✅ docs
6cf37fa2b2ea6dc03f52c1a729065b58c6d1aea0 @arhimondr #14448 ✅ rn ✅ docs
78fd3b1f7b9aa763bc41623157158cb94b8d15bc @arhimondr #14448 ✅ rn ✅ docs
2cf1d4a88936a6368525c6f219068e2ad2ae68de @arhimondr #14448 ✅ rn ✅ docs
12b27dee50babff011411622ab92ace63e0fb747 @arhimondr #14448 ✅ rn ✅ docs
5f7f648993f94e35428aa16ce82070c4e8ca4aa8 @arhimondr #14448 ✅ rn ✅ docs
dd8dee5e9ace608c6be5cf06878ff75240577650 @arhimondr #14448 ✅ rn ✅ docs
911d7ca3d0b842d9ae40003b1b2b5d34388d2b80 @arhimondr #14225 ✅ rn ➖ docs (#14599)
f76fa46d172ff8d7b22b9340c2c76921232d50d9 @arhimondr #14225 ✅ rn ➖ docs (#14599)
ca4cf8f93f0e654eacd1b07fa21c74d0c7df42f6 @arhimondr #14225 ✅ rn ➖ docs (#14599)

## 7 Oct 2022 ✅ 

a2fe9f4d5e7d81f7fdda9c65fb8bc8c771fc2338 @findepi #14492 ✅ rn ✅ docs
6dab3e688bed4c91ede91acff1015de65a3cb11e @findepi #14492 ✅ rn ✅ docs
4c930e31b15598a29d14c47a2ec2e989fe53db38 @ebyhr #14511 ✅ rn ✅ docs
2ea06f3b8297fc9f08d2363db4958372f3adc67e @ebyhr #14512 ✅ rn ✅ docs
ec4374ff4ff4e4e75f2af92a2d3df59d5022811b @ebyhr #14498 ✅ rn ✅ docs
4ed05ccf5ce4cf61dff729b8545ed59a022115ff @findepi #14483 ✅ rn ✅ docs
ae11dd63d192ae7a70b82144961c160467195d91 @findepi #14489 ✅ rn ✅ docs
be5aa67d8587dc9bd9e9a22aa0f6ae8b02fba39b @findepi #14489 ✅ rn ✅ docs
889066d0a722b8795014553fc15fe1d9731502bd @findepi #14489 ✅ rn ✅ docs
1e89fb20cda8b7c258112093801bbf991b5419e6 @findepi #14489 ✅ rn ✅ docs
0120169f36bada931b1cc06ed016d2492de3953d @sopel39 #14466 ✅ rn ✅ docs
e9c77dd36ad32343d3597b631113192fb9172f05 @ebyhr #14475 ✅ rn ✅ docs
39e18ec66cbfc5b1e707179433029cd4a38dd287 @findepi #14520 ✅ rn ✅ docs
0321394db764561c07db1558da3ab65f865dda29 @findepi #14522 ✅ rn ✅ docs
d49e1b399c688d02dfec1354e0da60865088a1de @ebyhr #14490 ✅ rn ✅ docs
b44180db85284bb703c8e8728823bd513ab0b6c9 @ebyhr #14510 ✅ rn ✅ docs
326b78668c1a4489fa7204e184775241946c89a1 @ebyhr #14510 ✅ rn ✅ docs
1a2a2ef2181c63919749f61a708e159a85c47fee @ebyhr #14513 ✅ rn ✅ docs
f9b3b0e7f17bb138f7a1475fa5df1dfba2252d8c @ebyhr #14180 ✅ rn ✅ docs
171959b2fbe318399f528b9db80b7c272cd25b10 @ebyhr #14524 ✅ rn ✅ docs

## 8 Oct 2022 ✅ 

4302296cdb8f3ed87223245c65b9d705b05446cd @kokosing #14191 ✅ rn ✅ docs
76395ef791b90d8ce3dc7eaad80164f71970372c @martint #14090 ✅ rn ✅ docs
511e5d222417cbca74ffcf3f6231e61c99057500 @martint #14090 ✅ rn ✅ docs
037434261523095665d686782e3a5cfed0308bef @martint #14090 ✅ rn ✅ docs
da19853582acdd361f588c458917006bdfa97a19 @electrum #10984 ✅ rn ✅ docs
091a3547d7b5eaf097fc6f13b48cc52cee4765f5 @electrum #10984 ✅ rn ✅ docs
f3a13230ddacb2c5dd5d8b6ad4af7a13f07db256 @electrum #10984 ✅ rn ✅ docs
74602a6d4588bc5acd356a87fdd2669d810c1279 @electrum #10984 ✅ rn ✅ docs
5463b037a0ead5fa562261a04c4fef94f7d2a2fc @electrum #10984 ✅ rn ✅ docs
c43967ad1e911879d7e00bf8db85decacb3597ca @electrum #10984 ✅ rn ✅ docs

## 9 Oct 2022 ✅ 

5b881720fbfb141f598a9fbc8a76009ed19c279b @electrum #14534 ✅ rn ✅ docs
bc59aca80988bbebfa9c8f948bc52c24c02a31de @electrum #14534 ✅ rn ✅ docs
c9deaf6e0e5f710e6fa4964caed5844be77b1253 @electrum #14534 ✅ rn ✅ docs
4ae6fcda9ff74c2a4943af62e7dd7ebcdff93359 @martint #13015 ✅ rn ✅ docs

## 10 Oct 2022 ✅ 

1052d1bef33978856c801ce0fcdc704009e55338 @hashhar #14177 ✅ rn ✅ docs
481c21b7914045f371a94716949682e7179636c5 @hashhar #14536 ✅ rn ✅ docs
17bf26ced3a6379d11c83cab669d733368e448d5 @findepi #14470 ✅ rn ✅ docs
96a009eda3ea13d832e4500a89df2dccc0e6e6cf @findepi #14523 ✅ rn ✅ docs
2c68ff24463f95179edb7b28ffae32cf40dbdb0f @findepi #14386 ✅ rn ✅ docs
4674cc52bae5b2aa7c07fe64c0ef2693341a5af1 @hashhar #14272 ✅ rn ✅ docs
ba602d31840c7446b5870ef06beb1755b27e35fd @findepi #14548 ✅ rn ✅ docs
a2104a3be956efdceae8f782c9d4bca3989a086c @sopel39 #14395 ✅ rn ✅ docs
426e759456001a4b58234d95aae03da6eb0e1ec0 @sopel39 #14395 ✅ rn ✅ docs
e628403f558919b7fc8a83b9b699130e9d7fbbdb @kokosing #14541 ✅ rn ✅ docs
3cc0058f82a71f15cef53ac160d043cb47cea6e3 @ebyhr #14494 ✅ rn ✅ docs
cf7ac01b9b0a0b1a56514ba136d5ceadaad5d022 @dain #14555 ✅ rn ✅ docs
0e79dfaa586f38e64247775e9caba887f4093ef8 @ebyhr #14314 ✅ rn ✅ docs
04e650aa35ae7b7651b4d7d226ab21cf6ff7333a @ebyhr #14314 ✅ rn ✅ docs
a9480bd1e5fbdad87b62f33c4d9323a021874963 @ebyhr #14314 ✅ rn ✅ docs

## 11 Oct 2022 ✅ 

8e4153dd951638a5066e02da69ff3a544f53d980 @kokosing #14518 ✅ rn ✅ docs
0c815b18566453443e769a6db3dedbe5bb1f98a0 @findepi #14550 ✅ rn ✅ docs
e14964754ca31ef57900c1ffe4973da27d8ec1b4 @findepi #14502 ✅ rn ✅ docs
bdd313248eeb294557dd12e9d6f059077ada816a @findepi #14502 ✅ rn ✅ docs
3490d6f90166f8199588a4f254a344b428dfc7d2 @ebyhr #14558 ✅ rn ✅ docs
f4f55f02e96e987109e4a3f43e64e322167b7034 @sopel39 #14396 ✅ rn ✅ docs
2496fac48c9353070ee7697b5f15591890e0d92b @sopel39 #14396 ✅ rn ✅ docs
9919c35150c47308794693686ac277037d029140 @findepi #14565 ✅ rn ✅ docs
09edb1ad447c50487da9b5431e8568e97313e965 @findepi #14565 ✅ rn ✅ docs
968c8741cdd79d168958b45e12a3ab711fe12469 @ebyhr #14389 ✅ rn ✅ docs
0c81592ab185e3f8f044ef0168bc7f1fef059eb1 @ebyhr #14389 ✅ rn ✅ docs
91345865c62e2fe06c1570f7872ffb7b858a045f @findepi #14571 ✅ rn ➖ docs (#14600)

## 12 October 2022

a8447634edb33af1637ca56c5281208d32fa58d5 @sopel39 #14559 ✅ rn ✅ docs
ee76797071237b66d8bff4d24d964dc23cc6b87e @ebyhr #13822 ✅ rn ✅ docs
3198397a928c70893ae77b3387681fbaab5c3292 @ebyhr #13822 ✅ rn ✅ docs
a87fc6366f242ac8591935841babf13c438f3c20 @ebyhr #13822 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-400.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+* Add support for [truncating tables](/sql/truncate). ({issue}`14494`)
+
+## Hudi connector
+
+* Hide Hive system schemas. ({issue}`14510`)
+
+## Iceberg connector
+
+* Prevent table corruption when changing a table fails due to inability to
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-400.md:18]: maybe 

```suggestion
* Prevent table corruption when changing a table fails due to an inability to
```

**Review by mosabua — APPROVED** [review]
One minor suggestion - otherwise looks good to go.

File: docs/src/main/sphinx/release/release-400.md [hunk 2]
```diff
@@ -9,6 +9,21 @@
...
 * Add support for [truncating tables](/sql/truncate). ({issue}`14494`)
 
+## Delta Lake connector
+
+* Prevent coordinator out-of-memory failure when querying a large number of
+  tables in a short period of time. ({issue}`14571`)
+
+## Hive connector
+
+* Reduce memory usage when scanning a large number of partitions, and add the
```
Comment (martint) [inline docs/src/main/sphinx/release/release-400.md:19]: > ... *__that__* can be loaded

File: docs/src/main/sphinx/release/release-400.md [hunk 3]
```diff
@@ -9,6 +9,21 @@
...
+  `hive.max-partitions-for-eager-load` configuration property to manage the
+  number of partitions which can be loaded into memory. ({issue}`14225`)
+* Increase the default value of the `hive. max-partitions-per-scan`
```
Comment (martint) [inline docs/src/main/sphinx/release/release-400.md:22]: Extra space after `hive.`

File: docs/src/main/sphinx/release/release-400.md [hunk 4]
```diff
@@ -9,6 +9,21 @@
...
+  configuration property to 1,000,000 from 100,000. ({issue}`14225`)
+* Utilize the `hive.metastore.thrift.delete-files-on-drop` configuration
```
Comment (martint) [inline docs/src/main/sphinx/release/release-400.md:24]: What does this mean? Did it exist but it was not being used? That would be a bug.

Comment (colebow) [reply docs/src/main/sphinx/release/release-400.md:24]: It existed but was only being used for table drops, not partition drops.

File: docs/src/main/sphinx/release/release-400.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+* Prevent coordinator out-of-memory failure when querying a large number of
+  tables in a short period of time. ({issue}`14571`)
+
+## Hive connector
+
+* Reduce memory usage when scanning a large number of partitions, and add the
+  `hive.max-partitions-for-eager-load` configuration property to manage the
+  number of partitions that can be loaded into memory. ({issue}`14225`)
+* Increase the default value of the `hive.max-partitions-per-scan`
+  configuration property to 1,000,000 from 100,000. ({issue}`14225`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-400.md:23]: I know that formatting makes it more readable .. this sort of format wont work in the actual property I think .. so should we leave it or change to code highlighting like so

```
`1000000` from `100000`
```

File: docs/src/main/sphinx/release/release-400.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+## Hive connector
+
+* Reduce memory usage when scanning a large number of partitions, and add the
+  `hive.max-partitions-for-eager-load` configuration property to manage the
+  number of partitions that can be loaded into memory. ({issue}`14225`)
+* Increase the default value of the `hive.max-partitions-per-scan`
+  configuration property to `1000000` from `100000`. ({issue}`14225`)
+* Utilize the `hive.metastore.thrift.delete-files-on-drop` configuration
+  property when dropping partitions and tables, previously it was only used when
```
Comment (martint) [inline docs/src/main/sphinx/release/release-400.md:25]: Run-on sentence. Use a period and make it two separate sentences.

**Comment by martint** [comment]
Tested locally

**Pull request merged**